ddtrace/tracer/log.go

```diff
@@ -47,18 +46,16 @@ type startupInfo struct {
 	GlobalService         string            `json:"global_service"`          // Global service string. If not-nil should be same as Service. (#614)
 }
 
-// checkEndpoint tries to connect to the URL specified by endpoint.
-// If the endpoint is not reachable, checkEndpoint returns an error
-// explaining why.
-func checkEndpoint(endpoint string) error {
-	req, err := http.NewRequest("POST", endpoint, nil)
-	if err != nil {
-		return fmt.Errorf("cannot create http request: %v", err)
-	}
-	_, err = defaultClient.Do(req)
+// checkEndpoint tries to send an empty payload through the transport.
+// If the transport fails, it returns the cause as an error.
+func checkEndpoint(t transport) error {
+	p := newPayload()
+	p.updateHeader()
+	b, err := t.send(p)
```

Comment:
This looks hacky, especially the `updateHeader` call makes no sense and looks out of place. I liked the previous code more. It simply needs a body added, the empty array: `bytes.NewReader([]byte{0x90})`) and a header to the request.